tests/versioned/apollo-federation/segments.test.js

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright 2020 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const { executeQuery, executeQueryBatch } = require('../../test-client')
+
+const ANON_PLACEHOLDER = '<anonymous>'
+const OPERATION_PREFIX = 'GraphQL/operation/ApolloServer'
+const EXTERNAL_PREFIX = 'External'
+
+const { setupFederatedGatewayServerTests } = require('./federated-gateway-server-setup')
+
+setupFederatedGatewayServerTests({
+  suiteName: 'fastify segments',
+  createTests: createFederatedSegmentsTests
+})
+
+/**
+ * Creates a set of standard segment naming and nesting tests to run
+ * against express-based apollo-server libraries.
+ * It is required that t.context.helper and t.context.serverUrl are set.
+ * @param {*} t a tap test instance
+ */
+function createFederatedSegmentsTests(t, frameworkName) {
+  const TRANSACTION_PREFIX = `WebTransaction/${frameworkName}/POST`
+
+  t.test('should nest sub graphs under operation', (t) => {
+    const { helper, serverUrl } = t.context
+
+    const query = `query {
+      libraries {
+        branch
+        booksInStock {
+          isbn,
+          title,
+          author
+        }
+        magazinesInStock {
+          issue,
+          title
+        }
+      }
+    }`
+
+    helper.agent.on('transactionFinished', (transaction) => {
+      if (transaction.forceIgnore) {
+        return
+      }
```

Comment:
This is ignoring the sub-graph transactions. See the related comment in the gateway server setup file.